Move front-page template to page-home instead.

kjellr opened the pull request:
Fixes https://github.com/WordPress/twentytwentytwo/issues/60. This PR renames `front-page.php` to `page-home.php` so that the theme will respect the user's front page settings.

kjellr committed:
Move front-page template to home instead.

Sandstromer commented on block-templates/home.html:
```diff
@@ -0,0 +1,55 @@
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p><a href="https://2022kjell.mystagingwebsite.com/blog/" data-type="page" data-id="8">View All</a></p>
```
Not sure if leaving this hardcoded link is intentional.

kjellr committed:
Remove the home.html template entirely for now.

kjellr committed:
Show an even number of posts per page by default in index

kjellr commented:
I reworked this a bit (and updated the description). Originally I tried having a `home.html` template that had the large black header, but that [forced a lot of extra scrolling](https://cloudup.com/cTpeH0BHWdt) when you paginated. I might add a `home.html` template later on, but I'll need to redesign it so this is smoother.

This new approach forces folks to either name their homepage with the `home` slug, or to manually select the `page-home` template for their homepage. But I think that's more or less what the template hierarchy dictates. Maybe we can use starter content to make that easier.

jffng commented on block-templates/index.html:
```diff
@@ -1,7 +1,7 @@
 
 <!-- wp:group {"style":{"spacing":{"padding":{"right":"max(1.25rem, 5vw)","left":"max(1.25rem, 5vw)"}}}} -->
-<div class="wp-block-group" style="padding-right:max(1.25rem, 5vw);padding-left:max(1.25rem, 5vw)"><!-- wp:query {"queryId":1,"query":{"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","sticky":"","perPage":10},"tagName":"main","displayLayout":{"type":"flex","columns":3},"layout":{"inherit":true}} -->
+<div class="wp-block-group" style="padding-right:max(1.25rem, 5vw);padding-left:max(1.25rem, 5vw)"><!-- wp:query {"queryId":1,"query":{"offset":0,"postType":"post","categoryIds":[],"tagIds":[],"order":"desc","orderBy":"date","author":"","search":"","sticky":"","perPage":12},"tagName":"main","displayLayout":{"type":"flex","columns":3},"layout":{"inherit":true}} -->
```
This may be a separate issue, but is there a way to get the query to respect the Reading settings for max blog posts to show?

<img width="486" alt="Screen Shot 2021-10-08 at 11 49 09 AM" src="https://user-images.githubusercontent.com/5375500/136586499-e1a3c7c9-2161-4f63-a886-ccfc7558e861.png">

I think it's 10 by default.

kjellr replied:
Yeah, I _think_ if you leave that out it will respect this setting. But we definitely want to have the default be a multiple of three though since we're showing things in three columns. It seems like setting that here is the best way to do this for a block theme. 🤷‍♂️
